File: app/src/api/stt.js
import { request } from "./client.js";

// → { text: string }
export const transcribeVoice = (audioBlob, filename = "voice.webm") => {
  const fd = new FormData();
  fd.append("audio", audioBlob, filename);
  return request("/stt", { method: "POST", body: fd });
};

File: gaz-server/src/routes/stt.py
"""POST /stt — voice query transcription."""

from fastapi import APIRouter, File, UploadFile

from ..schemas.rag import SttResponse
from ..services.stt_service import transcribe_voice

router = APIRouter()


@router.post("/stt", response_model=SttResponse)
async def stt(audio: UploadFile = File(...)) -> SttResponse:
    audio_bytes = await audio.read()
    text = await transcribe_voice(audio_bytes, audio.filename or "voice.webm")
    return SttResponse(text=text)
